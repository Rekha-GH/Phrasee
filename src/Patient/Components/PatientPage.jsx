import React from 'react';
import { Row, Col } from 'antd';
import PatientBlock from './PatientBlock';
import { useSelector } from 'react-redux';
import { getType } from '../Store/PatientSelector';
import { getTypeName } from '../Utils/helper';


const PatientPage = React.memo(() => {
    const types = useSelector(getType);
    return (
        <Row>
            <Col span={12} offset={6}>
                <h2>Patients</h2>
                {types.map((type) => <PatientBlock key={type} title={getTypeName(type)} type={type} />)}
            </Col>
        </Row>
    );
});

PatientPage.displayName = 'PatientPage';
export default PatientPage;