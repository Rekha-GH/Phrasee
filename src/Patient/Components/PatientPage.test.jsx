import React from 'react';
import toJson from 'enzyme-to-json';
import * as ReactRedux from 'react-redux';


import PatientPage from './PatientPage';
import { shallow } from '../../setupTests';

describe('PatientPage Tests', () => {
  const useSelectorMock = jest.spyOn(ReactRedux, 'useSelector');

  test('PatientPage renders correctly when types are not available', () => {
    useSelectorMock.mockImplementationOnce(() => []); 
    const wrapper = shallow(<PatientPage />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('PatientPage renders correctly when types are available', () => {
    useSelectorMock.mockImplementationOnce(() => ['a_level', 'b_level']);
    
    const wrapper = shallow(<PatientPage />);
    
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
