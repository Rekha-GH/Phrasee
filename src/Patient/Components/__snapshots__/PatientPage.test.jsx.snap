// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PatientPage Tests PatientPage renders correctly when types are available 1`] = `
<Row>
  <Col
    offset={6}
    span={12}
  >
    <h2>
      Patients
    </h2>
    <PatientBlock
      key="a_level"
      title="a level"
      type="a_level"
    />
    <PatientBlock
      key="b_level"
      title="b level"
      type="b_level"
    />
  </Col>
</Row>
`;

exports[`PatientPage Tests PatientPage renders correctly when types are not available 1`] = `
<Row>
  <Col
    offset={6}
    span={12}
  >
    <h2>
      Patients
    </h2>
  </Col>
</Row>
`;
